frontend/src/features/landing_page/components/PhotoGallery.tsx

```diff
@@ -1,37 +1,39 @@
 import SimpleButton, { ButtonVariant } from '@/components/Button/SimpleButton';
 import '@/components/Modals/ImageModal';
 import ImageModal from '@/components/Modals/ImageModal';
+import { getPublicFileList, getPublicImage } from '@/features/nextcloud_integration/api/getImages';
+import { ImageBlob } from '@/features/nextcloud_integration/components/ImageBlob';
+import { useQuery } from '@tanstack/react-query';
 import { useState } from 'react';
 
+async function getPublicImages(imagePaths: Array<string>, publicShareToken: string) {
+  return Promise.all(imagePaths.map((path) => getPublicImage(path, publicShareToken)));
+}
+
+function getImagesFromFileList(files: Array<string>) {
+  return files.filter((file) => {
+    // check file extension
+    const parts = file.split('.');
+    const extension = parts[parts.length - 1];
+    return ['png', 'jpg', 'jpeg', 'svg'].includes(extension);
```

Comment:
Is this needed? We can keep the shares purely with images? If it is needed: wouldn't mime-type filtering be better?

Reply:
See my comment below for a suggestion on how to only filter images from the XML.